internal/cmd/envs.go

```diff
@@ -117,3 +126,85 @@ coder envs --user charlie@coder.com ls -o json \
 		},
 	}
 }
+
+func createEnvCommand() *cobra.Command {
+	var (
+		org      string
+		img      string
+		tag      string
+		services []string
+	)
+
+	cmd := &cobra.Command{
+		Use:    "create [flags] [environment-name]",
+		Short:  "create a new environment.",
+		Args:   cobra.ExactArgs(1),
+		Hidden: true,
+		Long:   "Create a new environment under the active user.",
+		Example: `
+# create a new environment using default resource amounts
+coder envs create \
+	--image id-of-imported-image \
+	my-env-name
+
+See Flags section for default values.
+
+# create a new environment using custom resource amounts
+coder envs create \
+	--cores 4 \
+	--disk 100 \
+	--memory 8 \
+	--image id-of-imported-image \
+	--org id-of-existing-organization \
+	my-env-name
+
+# same command this time using short-hand flags for a more succinct command experience.
+coder envs create \
+	-c 4 \
+	-d 100 \
+	-m 8 \
+	-i id-of-imported-image my-env-name \
+	-o id-of-existing-organization \
+	my-env-name
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if img == "" {
+				return xerrors.New("image id unset")
+			}
+			// ExactArgs(1) ensures our name value can't panic on an out of bounds.
+			createReq := &coder.CreateEnvironmentRequest{
+				Name:     args[0],
+				ImageID:  img,
+				ImageTag: tag,
+			}
+			// We're explicitly ignoring errors for these because all of these flags
+			// have a non-zero-value default value set already.
+			createReq.CPUCores, _ = cmd.Flags().GetFloat32("cores")
+			createReq.MemoryGB, _ = cmd.Flags().GetFloat32("memory")
+			createReq.DiskGB, _ = cmd.Flags().GetInt("disk")
+			createReq.GPUs, _ = cmd.Flags().GetInt("gpus")
+
+			client, err := newClient()
+			if err != nil {
+				return err
+			}
+
+			env, err := client.CreateEnvironment(cmd.Context(), org, *createReq)
+			if err != nil {
+				return xerrors.Errorf("create environment: %w", err)
+			}
+			flog.Success("Successfully created environment %q", env.Name)
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&org, "org", "o", defaultOrg, "ID of the organization the environment should be created under.")
+	cmd.Flags().StringVarP(&tag, "tag", "t", defaultImgTag, "The particular tag of the image the environment will be based off of.")
+	cmd.Flags().Float32P("cores", "c", defaultCPUCores, "The number of cpu cores the environment should be provisioned with.")
+	cmd.Flags().Float32P("memory", "m", defaultMemGB, "GB of RAM an environment should be provisioned with.")
+	cmd.Flags().IntP("disk", "d", defaultDiskGB, "GB of disk storage an environment should be provisioned with.")
+	cmd.Flags().IntP("gpus", "g", defaultGPUs, "The number GPUs an environment should be provisioned with.")
+	cmd.Flags().StringSliceP("services", "s", services, "The services that the environment should be initialized with.")
+	cmd.Flags().StringVarP(&img, "image", "i", "", "ID of the image to base the environment off of.")
```

Comment:
I'm not a huge fan of us requiring random unique identifiers as arguments. The optimal solution here feels like the following:
```
--image-id <id of image>
--image <name of image>
```

Where we error if both are set.

Reply:
The image-id is a hard requirement for sending the create request. 

If we feel strongly about this, we'd have to add an endpoint to the enterprise API in order to retrieve an image by name so we can use the image-id in the create environment request.

The clilent will require the image-id either way.

Passing a name would be nice but I don't think taking an image id is out of the ordinary(Docker commands take container id's).

We could also have a `coder images ls` in the future that helps make obtaining these image id's more convenient.

Reply:
Yeah I was thinking we'd have something like `findEnv` for images. We've been pretty good so far about using human friendly values so let's stick to that before unhiding this. Makes sense to tackle that in a follow-up PR though.